test_runner/src/main/kotlin/ftl/util/Obfuscation.kt

```diff
@@ -0,0 +1,71 @@
+package ftl.util
+
+private const val LOWER_CASE_CHARS = "abcdefghijklmnopqrstuvwxyz"
+private const val UPPER_CASE_CHARS = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+private const val START_CONTEXT_KEY = ""
+
+interface Obfuscation {
+    fun obfuscate(input: String): String
+}
+
+class AndroidObfuscation : Obfuscation {
+    private val contextMapAndroid = mutableMapOf<String, MutableMap<String, String>>()
+    private val androidTestMethodSeparator = '#'
+    private val androidPackageSeparator = '.'
+
+    override fun obfuscate(input: String): String {
+        val obfuscatedPackageNameWithClass =
+            obfuscateAndroidPackageAndClass(input.split(androidTestMethodSeparator).first())
+
+        val obfuscatedMethodName = obfuscateMethodName(
+            methodName = input.split(androidTestMethodSeparator).last(),
+            context = contextMapAndroid.getOrPut(obfuscatedPackageNameWithClass) { mutableMapOf() }
+        )
+
+        return "$obfuscatedPackageNameWithClass$androidTestMethodSeparator$obfuscatedMethodName"
+    }
+
+    private fun obfuscateAndroidPackageAndClass(packageNameWithClass: String): String {
+        return packageNameWithClass
+            .split(androidPackageSeparator)
+            .fold(START_CONTEXT_KEY) { previous, next ->
```

Comment:
IMO, `START_CONTEXT_KEY` is redundant. `fold("")` is better than `fold(START_CONTEXT_KEY)` because `START_CONTEXT_KEY` forces to remember next name, and scroll to check what value it holds.